tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -165,7 +168,10 @@ private void OnBeginRequest(object sender, EventArgs eventArgs)
             {
                 // Dispose here, as the scope won't be in context items and won't get disposed on request end in that case...
                 scope?.Dispose();
-                Log.Error(ex, "Datadog ASP.NET HttpModule instrumentation error");
+                if (ex is not BlockException)
+                {
+                    Log.Error(ex, "Datadog ASP.NET HttpModule instrumentation error");
+                }
             }
```

Comment:
This should be the right spot to end this request (or redirect to a blocking screen), IMHO

Reply:
It is done by the `security.InstrumentationGateway.RaiseBlockingOpportunity` , as per design, we go through the `InstrumentationGateway`